app/src/apiClient.js

```diff
@@ -1,6 +1,16 @@
-export const getTasks = () => _get("/api/tasks");
+export const getPlayers = () => _get("/api/players");
 
-export const addTask = (name) => _post("/api/tasks", { name });
+export const getTopPlayers = () => _get("/api/players/top");
```

Comment:
So I remember what I really wanted to focus on was your API design, I think if you just had good CRUD on your players resource, you wouldn't need so many custom endpoints, e.g. just expose:

```
POST /api/players
GET /api/players/:id
GET /api/players  -> with support for a query param "top"
PUT (or PATCH) /api/players to allow you to update win/loss/tie counts on the players resource
```

The upside of this approach is it is a more idiomatic usage of REST the downside being it is harder to prevent the client from doing things you don't want it to do, e.g. if you don't want the client to increment wins by more than one, etc, that would be harder to enforce with the API I lay out, but you could still do it.

Let me think about some other ways you might be able to lay out this API to make it seem a bit more intuitive.

Reply:
As another option in-between what you have and really strict rest (purely CRUD on resources) would be:

```
POST /api/players (create player)
DELETE /api/players/:id (delete player)
POST /api/player/:id/win (create a "win resource" for this player, more details below)
POST /api/player/:id/loss (same as for the previous but for loss)
POST /api/player/:id/tie (same as for previous)
```

I kind of think you could either have separate "synthetic resources" for win/loss/tie which have a schema like:
{
playerOneId: id
playerTwoId: id
winnerId: id
isTie
}
```

Also instead of having three separate endpoints for tracking win, loss tie it might be fine just to have a single `/api/players/win` (no id) resource which takes the schema i describe above with all the information needed to record a single when, then you can do the sequence of BE/DB actions to update all the players involved with the information in the resource.

Hope that makes sense but let me know if I am being too vague!